### PR TITLE
fix(initramfs): Fix case where we download a non archive file

### PR DESCRIPTION
**Pull request by bradbeam**
When the previous changes went in for the Install() func, we download to the appropriate destination and return the sourceFile as the downloaded file. Since the sourceFile is the actual file and not an archive, it triggers the second match in the switch statement.
Because we dont have a destFile defined, we get a fun seg fault.

Signed-off-by: Brad Beam <brad.beam@b-rad.info>

### CODE REVIEW

File: internal/pkg/install/install.go [hunk 1]
```diff
@@ -66,54 +66,67 @@ func Install(data *userdata.UserData) (err error) {
...
+					log.Printf("Failed to close %s", sourceFile.Name())
 					return err
 				}
-			case strings.HasPrefix(sourceFile.Name(), "/"):
+
+				if err = os.Remove(sourceFile.Name()); err != nil {
+					log.Printf("Failed to remove %s", sourceFile.Name())
+					return err
+				}
+			case strings.HasPrefix(sourceFile.Name(), "/") && destFile != nil:
```
Comment (bradbeam) [inline internal/pkg/install/install.go]: Just to highlight, this is the actual change. The rest are just 'helpful' log messages.

File: internal/pkg/install/install.go [hunk 2]
```diff
@@ -66,54 +66,67 @@ func Install(data *userdata.UserData) (err error) {
...
-				return err
-			}
-			if err = os.Remove(sourceFile.Name()); err != nil {
-				return err
+				if err = sourceFile.Close(); err != nil {
+					log.Printf("Failed to close %s", sourceFile.Name())
+					return err
+				}
 			}
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 3]
```diff
@@ -166,8 +179,10 @@ func untar(tarball *os.File, dst string) error {
 				TypeDir     = '5' // Directory
 				TypeFifo    = '6' // FIFO node
 		*/
+
 		switch header.Typeflag {
 		case tar.TypeDir:
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 4]
```diff
@@ -166,8 +179,10 @@ func untar(tarball *os.File, dst string) error {
 				TypeDir     = '5' // Directory
 				TypeFifo    = '6' // FIFO node
 		*/
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 5]
```diff
@@ -188,6 +203,7 @@ func untar(tarball *os.File, dst string) error {
 				return err
 			}
 		case tar.TypeSymlink:
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 6]
```diff
@@ -66,54 +66,67 @@ func Install(data *userdata.UserData) (err error) {
...
-				return err
+				if err = sourceFile.Close(); err != nil {
+					log.Printf("Failed to close %s", sourceFile.Name())
+					return err
+				}
 			}
+
 		}
 	}
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 7]
```diff
@@ -66,54 +66,67 @@ func Install(data *userdata.UserData) (err error) {
...
 					return err
 				}
-			case strings.HasPrefix(sourceFile.Name(), "/"):
+
+				if err = os.Remove(sourceFile.Name()); err != nil {
+					log.Printf("Failed to remove %s", sourceFile.Name())
+					return err
+				}
+			case strings.HasPrefix(sourceFile.Name(), "/") && destFile != nil:
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 8]
```diff
@@ -66,54 +66,67 @@ func Install(data *userdata.UserData) (err error) {
...
-					if err = os.Symlink(link[1], filepath.Join(dest, link[0])); err != nil && !os.IsExist(err) {
-						return err
-					}
-				}
 			}
 
 			switch {
 			case strings.HasSuffix(sourceFile.Name(), ".tar") || strings.HasSuffix(sourceFile.Name(), ".tar.gz"):
-				log.Printf("Extracting %s\n", artifact)
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

File: internal/pkg/install/install.go [hunk 9]
```diff
@@ -49,7 +49,7 @@ func Install(data *userdata.UserData) (err error) {
 
 		// Extract artifact if necessary, otherwise place at root of partition/filesystem
 		for _, artifact := range urls {
-			log.Println(artifact)
+
```
Comment (andrewrynhard) [inline internal/pkg/install/install.go]: nit: extra line

**Pull request merged**